### PR TITLE
add: pass swagger js, url and docs favicon

### PR DESCRIPTION
**Pull request by temirgaliyev**
Hi,
there is some changes to be able to fix swagger-ui's url on FastApi class creation
That might help to avoid issues like https://github.com/swagger-api/swagger-ui/issues/6254 and hacks like dadd6650ed0418722ff3e39364fed53460ec62ea in the future from the developers perspective
This might also be helpful if someone don`t want to use jsdelivr's cdn

### CODE REVIEW

File: fastapi/applications.py [hunk 1]
```diff
@@ -42,6 +42,9 @@ def __init__(
         default_response_class: Type[Response] = JSONResponse,
         docs_url: Optional[str] = "/docs",
         redoc_url: Optional[str] = "/redoc",
+        swagger_ui_js_url: Optional[str] = "https://cdn.jsdelivr.net/npm/swagger-ui-dist@3.30.0/swagger-ui-bundle.js",
```
Comment (Kludex) [inline fastapi/applications.py:45]:
```suggestion
        swagger_ui_js_url: str = "https://cdn.jsdelivr.net/npm/swagger-ui-dist@3.30.0/swagger-ui-bundle.js",
```

File: fastapi/applications.py [hunk 2]
```diff
@@ -42,6 +42,9 @@ def __init__(
...
+        swagger_ui_css_url: Optional[str] = "https://cdn.jsdelivr.net/npm/swagger-ui-dist@3.30.0/swagger-ui.css",
```
Comment (Kludex) [inline fastapi/applications.py:46]:
```suggestion
        swagger_ui_css_url: str = "https://cdn.jsdelivr.net/npm/swagger-ui-dist@3.30.0/swagger-ui.css",
```

File: fastapi/applications.py [hunk 3]
```diff
@@ -42,6 +42,9 @@ def __init__(
...
+        swagger_ui_favicon_url: Optional[str] = "https://fastapi.tiangolo.com/img/favicon.png",
```
Comment (Kludex) [inline fastapi/applications.py:47]:
```suggestion
        swagger_ui_favicon_url: str = "https://fastapi.tiangolo.com/img/favicon.png",
```

File: fastapi/openapi/docs.py [hunk 4]
```diff
@@ -9,9 +9,9 @@ def get_swagger_ui_html(
     *,
     openapi_url: str,
     title: str,
-    swagger_js_url: str = "https://cdn.jsdelivr.net/npm/swagger-ui-dist@3.30.0/swagger-ui-bundle.js",
-    swagger_css_url: str = "https://cdn.jsdelivr.net/npm/swagger-ui-dist@3.30.0/swagger-ui.css",
-    swagger_favicon_url: str = "https://fastapi.tiangolo.com/img/favicon.png",
+    swagger_js_url: Optional[str] = "https://cdn.jsdelivr.net/npm/swagger-ui-dist@3.30.0/swagger-ui-bundle.js",
```
Comment (Kludex) [inline fastapi/openapi/docs.py:12]:
```suggestion
    swagger_js_url: str = "https://cdn.jsdelivr.net/npm/swagger-ui-dist@3.30.0/swagger-ui-bundle.js",
```

File: fastapi/openapi/docs.py [hunk 5]
```diff
@@ -9,9 +9,9 @@ def get_swagger_ui_html(
...
+    swagger_css_url: Optional[str] = "https://cdn.jsdelivr.net/npm/swagger-ui-dist@3.30.0/swagger-ui.css",
```
Comment (Kludex) [inline fastapi/openapi/docs.py:13]:
```suggestion
    swagger_css_url: str = "https://cdn.jsdelivr.net/npm/swagger-ui-dist@3.30.0/swagger-ui.css",
```

File: fastapi/openapi/docs.py [hunk 6]
```diff
@@ -9,9 +9,9 @@ def get_swagger_ui_html(
...
+    swagger_favicon_url: Optional[str] = "https://fastapi.tiangolo.com/img/favicon.png",
```
Comment (Kludex) [inline fastapi/openapi/docs.py:14]:
```suggestion
    swagger_favicon_url: str = "https://fastapi.tiangolo.com/img/favicon.png",
```

**Review by Kludex — COMMENTED** [review]
I haven't checked the MR per se... But the fields are not `Optional`, you're adding a default value that is `str`, the `Optional[X]` as a shorthand for `Union[X, None]`.

**Comment by temirgaliyev** [comment]
I don't really like how I hardcoded url's in both `fastapi/applications.py` and `fastapi/openapi/docs.py`, but `get_swagger_ui_html` is also used in `docs_src/extending_openapi/tutorial002.py` and `tests/test_local_docs.py` and changing these files could be an even worse idea.

**Comment by tiangolo** [comment]
Thanks for your interest!

Currently, it's fairly easy to add the docs UI manually with any level of customization wanted as described here: https://fastapi.tiangolo.com/advanced/extending-openapi/#include-the-custom-docs

And the number of possible parameters and configs to pass to `FastAPI()` could be enormous if I tried to support all the possible configurations, and I prefer to recommend customizing the docs UI directly.

So I'm gonna pass on this one. But thanks for your effort! :cake: :coffee:

**Pull request closed without merging**